Lib/test/test_traceback.py

```diff
@@ -3124,10 +3124,13 @@ def test_smoke_user_exception(self):
         class MyException(Exception):
             pass
 
-        self.do_test_smoke(
-            MyException('bad things happened'),
-            ('test.test_traceback.TestTracebackException.'
-             'test_smoke_user_exception.<locals>.MyException'))
+        if __name__ == '__main__':
+            expected = ('TestTracebackException.'
+                        'test_smoke_user_exception.<locals>.MyException')
+        else:
+            expected = ('test.test_traceback.TestTracebackException.'
+                        'test_smoke_user_exception.<locals>.MyException')
```

Comment:
```suggestion
        if __name__ == '__main__':
            expected = MyException.__qualname__
        else:
            expected = f'{__name__}.{MyException.__qualname__}'
```

Reply:
I would prefer to have test data as an explicit string. But, this is not a strong opinion :)

Reply:
> I would prefer to have test data as an explicit string. But, this is not a strong opinion :)

Oh, then let's leave it as it is.